src/libraries/System.Private.CoreLib/src/System/Version.cs

```diff
@@ -354,6 +354,15 @@ public static bool TryParse([NotNullWhen(true)] string? input, [NotNullWhen(true
 
         private static bool TryParseComponent(ReadOnlySpan<char> component, string componentName, bool throwOnFailure, out int parsedComponent)
         {
+            if (component.Length == 1)
+            {
+                var value = component[0] - '0';
```

Comment:
Explicit types are used here:
```suggestion
                int value = component[0] - '0';
```